hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java

```diff
@@ -189,6 +190,9 @@ public ClientNetworkConfig setConnectionTimeout(int connectionTimeout) {
      * @return configured {@link com.hazelcast.client.config.ClientNetworkConfig} for chaining
      */
     public ClientNetworkConfig addAddress(String... addresses) {
+        for (String address : addresses) {
+            checkHasText(address, "member must contain text");
+        }
```

Comment:
Could we add a proper IP/hostname validation, maybe with URL/URI class?

Reply:
see https://github.com/hazelcast/hazelcast/issues/11242#issuecomment-327823524
URI/URL classes are falling back to identifying as hostname if not a ipv4/ipv6. We can say  that is not an IP, but we can't say that it is not a valid hostname. And that is not enough. 
And also we don't want to get into hostname resolution also to validate hostname is valid. 
If is there a way that I don't know, I am open to suggestions.

Reply:
the checkHasText doesn't trim. So if somebody feeds spaces, tabs or linefeeds, these are not detected.

Reply:
I added the trim.